ospf6d/ospf6_asbr.c

```diff
@@ -67,9 +68,343 @@ static void ospf6_asbr_redistribute_unset(struct ospf6 *ospf6,
 #endif
 
 unsigned char conf_debug_ospf6_asbr = 0;
+unsigned char conf_debug_ospf6_summary = 0;
 
 #define ZROUTE_NAME(x) zebra_route_string(x)
 
+static void summary_address_originate(struct summary_address *sum)
+{
+	struct ospf6 *o = sum->ospf6;
+	struct ospf6_area *oa;
+	struct listnode *lnode;
+	struct ospf6_route *route;
+	struct ospf6_external_info *info;
+
+	/* Check if we already originated the route. */
+	if (sum->flags & SUMMARY_ADDRESS_F_ORIGINATED)
+		return;
+
+	/* Make sure it was not created. */
+	if (sum->route) {
+		SET_FLAG(sum->flags, SUMMARY_ADDRESS_F_ORIGINATED);
+		zlog_warn("%s: summary route %pFX already exists", __func__,
+			  &sum->node->p);
+		return;
+	}
+
+	if (IS_OSPF6_DEBUG_SUMMARY)
+		zlog_debug("Advertise summary route %pFX", &sum->node->p);
+
+	/* Create summary route and save it. */
+	route = ospf6_route_create();
+	route->type = OSPF6_DEST_TYPE_NETWORK;
+	/* Use big distance so to not shadow a legitimate route. */
+	route->path.cost = 255;
```

Comment:
Do we have a default path cost for summary ospf routes? I'm not certain 255 is the right number here?